plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php

```diff
@@ -104,7 +128,7 @@ public function output_errors() {
 			echo '</div>';
 
 			// Clear.
-			delete_option( 'woocommerce_meta_box_errors' );
+			delete_option( self::ERROR_STORE );
```

Comment:
This is still a problem, as another request can pick up the error message and deletes the option before the originating request gets to it. When I tested this, it's rather easy to trigger this by putting a breakpoint just after update option from tab/request 1, then opening another tab from the same admin and seeing the error on an unrelated screen and no error in tab 1. I wonder if we could fix this by outputting the error only if referer is the same (and storing it along with the message in the option table)? Maybe an idea for future improvement.

Reply:
Agree [(see also this comment)](https://github.com/woocommerce/woocommerce/pull/32540#issuecomment-1108826030).